src/clims/models/file.py

```diff
@@ -1,27 +1,42 @@
 from __future__ import absolute_import
-
+import os
 from django.db import models
-
+from tempfile import NamedTemporaryFile
 from sentry.db.models import FlexibleForeignKey, Model, sane_repr
 from openpyxl import load_workbook
 
 
-class StructuredFileMixin(object):
+class MultiFormatFile(object):
     """
-    A helper mixin for allowing plugin developers to easily parse files to common formats
-    without having to import requirements etc.
+    Wraps an organization file in order to generate any file format from it.
+    This functionality is wrapped in a separate class because excel handling
+    needs a temporary file that needs to be removed after usage.
     """
 
-    def as_excel(self, temp_file, read_only=True):
-        """
-        Returns the file as an excel file. The caller has the responsibility to
-        delete the temp file after usage!
-        """
-        with open(temp_file.name, 'wb') as f:
-            for _, chunk in enumerate(self.file.getfile()):
+    def __init__(self, organization_file):
+        self.temp_file = None
```

Comment:
A though there - perhaps we should make all the fields except `name` with `_`, since I'm guessing that they are not supposed to be accessed by the library user?

Reply:
Yes, I agree